### ISSUE TITLE
Pydantic type (DirectoryPath) failing to instantiate via Hypothesis plugin

### ISSUE THREAD

**Issue by rtbs-dev** [pydantic/pydantic#2694]
I believe the pydantic types for DirectoryPath and FilePath  ([these?](https://github.com/samuelcolvin/pydantic/blob/5261fd05a0374b84ce2602d45990baf480fa2417/pydantic/types.py#L681)) need to be updated to add a `_flavour` attribute for proper interface with Hypothesis. 

## Checks

* [x] I added a descriptive title to this issue
* [x] I have searched (google, github) for similar issues and couldn't find anything
* [x] I have read and followed [the docs](https://pydantic-docs.helpmanual.io/) and still think this is a bug

# Version Info

Output of `python -c "import pydantic.utils; print(pydantic.utils.version_info())"`:
```

             pydantic version: 1.8.1
            pydantic compiled: True
                 install path: /home/tbsexton/miniconda3/envs/ddrp/lib/python3.8/site-packages/pydantic
               python version: 3.8.8 | packaged by conda-forge | (default, Feb 20 2021, 16:22:27)  [GCC 9.3.0]
                     platform: Linux-5.4.0-72-generic-x86_64-with-glibc2.10
     optional deps. installed: ['typing-extensions']

```

# Setup:
 
```py
from pydantic import BaseModel,DirectoryPath
from hypothesis import given,strategies as st


class Publication(BaseModel):
    folder: DirectoryPath

@given(st.builds(Publication))
def test_pub_dim(inst: Publication):
    print(inst.schema())
    pass

test_pub_dim()
```
# Bug:
```
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-15-45fd095f4e05> in <module>
     11     pass
     12 
---> 13 test_pub_dim()

<ipython-input-15-45fd095f4e05> in test_pub_dim()
      7 
      8 @given(st.builds(Publication))
----> 9 def test_pub_dim(inst: Publication):
     10     print(inst.schema())
     11     pass

    [... skipping hidden 1 frame]

~/miniconda3/envs/ddrp/lib/python3.8/pathlib.py in __new__(cls, *args, **kwargs)
   1039         if cls is Path:
   1040             cls = WindowsPath if os.name == 'nt' else PosixPath
-> 1041         self = cls._from_parts(args, init=False)
   1042         if not self._flavour.is_supported:
   1043             raise NotImplementedError("cannot instantiate %r on your system"

~/miniconda3/envs/ddrp/lib/python3.8/pathlib.py in _from_parts(cls, args, init)
    680         # right flavour.
    681         self = object.__new__(cls)
--> 682         drv, root, parts = self._parse_args(args)
    683         self._drv = drv
    684         self._root = root

~/miniconda3/envs/ddrp/lib/python3.8/pathlib.py in _parse_args(cls, args)
    673                         "object returning str, not %r"
    674                         % type(a))
--> 675         return cls._flavour.parse_parts(parts)
    676 
    677     @classmethod

AttributeError: type object 'DirectoryPath' has no attribute '_flavour'

```

# Expectation
Per the docs: 
> From Pydantic v1.8 and Hypothesis v5.29.0, Hypothesis will automatically load support for custom types like PaymentCardNumber and PositiveFloat, so that the st.builds() and st.from_type() strategies support them without any user configuration.

The closest thing I could find was [this discussion on stackoverflow](https://stackoverflow.com/a/29851079):

> Here is the definition of the Path class. It does something rather clever. Rather than directly returning an instance of Path from its __new__(), it returns an instance of a subclass, but only if it's been invoked directly as Path() (and not as a subclass).
>
> Otherwise, it expects to have been invoked via either WindowsPath() or PosixPath(), which both provide a _flavour class attribute via multiple inheritance. You must also provide this attribute when subclassing. You'll probably need to instantiate and/or subclass the _Flavour class to do this. This is not a supported part of the API, so your code might break in a future version of Python.
>
> TL;DR: This idea is fraught with peril, and I fear that my answers to your questions will be interpreted as approval rather than reluctant assistance.

It seems that `pathlib.Path` is a bit wonky in how it manages Posix/Windows compatibility. Not sure how this would impact your other uses of Path though!

**Comment by RileyMShea** [comment]
I am also experience this.

CPython 3.9.5 on Ubuntu 20.04
Pydantic v1.8.2
